server/helpers/express-utils.ts

```diff
@@ -125,6 +126,34 @@ function getCountVideos (req: express.Request) {
   return req.query.skipCount !== true
 }
 
+// helpers added in server.ts and used in subsequent controllers used
+const apiResponseHelpers = (req, res: express.Response, next = null) => {
+  res.fail = (options) => {
+    const { data, status, message, title, type, docs, instance } = {
+      data: null,
+      ...options,
+      status: options.status || HttpStatusCode.BAD_REQUEST_400
+    }
+
+    const extension = new ProblemDocumentExtension({
+      ...data,
+      docs: docs || res.docs
```

Comment:
I don't really like to put the API doc URL in `res.docs`. I think your can build a function that automatically assign the doc depending on `req.originalUrl` and `req.method` if `docs` is not specified in `options`.

Reply:
> I think your can build a function that automatically assign the doc depending on `req.originalUrl` and `req.method` if `docs` is not specified in `options`.

`req` is often not passed down to where the response is put to fail.